src/components/Block/Block.styles.ts

```diff
@@ -0,0 +1,64 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import { BlockProps } from './Block';
+import { ellipsisStyle } from '../../util/styles';
+import { GutterSize } from '../FixedGrid/enums';
+import { textColors, fontSizes, lineHeights } from '../Text/Text.styles';
+import { IRawStyle } from '@uifabric/styling';
+
+export const gutterSize: { [size in GutterSize]: string } = {
+  none: '0',
+  xSmall: '0.4rem',
+  small: '0.8rem',
+  medium: '1.2rem',
+  large: '1.6rem',
+  xLarge: '2rem',
+  xxLarge: '2.4rem',
+};
+
+const getMarginTop = (topSpacing?: GutterSize, push?: number) => {
+  if (push) {
+    // For negative push, use top margin to "pull" it up
+    return push < 0 ? `${push / 10}rem` : undefined;
```

Comment:
i think the logic needs to handle the case when negative push and topSpacing are both provided:

```
const margin = topSpacing ? gutterSize[topSpacing] : 0;
if (push < 0) {
    return topSpacing + push; // adds the negative push value
}
return margin | undefined;
```

Reply:
From what I can tell, the old code would let a negative push win, and ignore the topSpacing.
Are you suggesting that the right thing to do would be to add the two?  If so, that's doable, but a bit painful since the gutterSize object contains strings like '1rem'.